lib/livebook/config.ex

```diff
@@ -705,12 +705,21 @@ defmodule Livebook.Config do
       provider ->
         with [type, key] <- String.split(provider, ":", parts: 2),
              %{^type => module} <- identity_provider_type_to_module() do
-          {module, key}
+          {:zta, module, key}
         else
           _ -> abort!("invalid configuration for identity provider given in #{env}")
         end
     end
   end
 
+  # Allow for this function to be tested more easily
+  defp get_system_env_or_test_provider(env) do
+    case {System.get_env(env), env} do
+      {nil, "TEST_IDENTITY_" <> provider} -> provider
+      {nil, _} -> nil
+      {provider, _} -> provider
+    end
+  end
```

Comment:
Instead of doing this, have a `parse_identity_provider!` function that already receives the value, this way you don't need to mess with env vars.

Reply:
:thinking: let me :runner: with this for a minute

Reply:
* created `identity_provider!/0` that defines t he environment checking
* created `identity_provider!/1`  with pattern matching instead of case statement

I think this might have been what you were thinking, otherwise I can just do the `with_env` like the other tests if we want. not sure what you might want here.

Reply:
`with_env` is definitely the way to go :)